src/test/java/org/kiwiproject/registry/consul/server/ConsulRegistryServiceIntegrationTest.java

```diff
@@ -0,0 +1,199 @@
+                    .withStatus(ServiceInstance.Status.UP);
+
+            var registeredInstance = service.register(serviceInstance);
+
+            assertThat(registeredInstance).isNotSameAs(serviceInstance);
+
+            var storedInstance = service.registeredService.get();
+            assertThat(storedInstance).isNotNull();
+            assertThat(storedInstance.getInstanceId()).isNotBlank();
+
+            assertThat(consul.catalogClient().getService(serviceInstance.getServiceName()).getResponse()).isNotEmpty();
+        }
+
+        @Test
+        void shouldRetryRegisterAndThrowExceptionIfAllTriesExpire() {
+            var badConsul = mock(Consul.class);
+            var serviceToFail = new ConsulRegistryService(badConsul, new ConsulRegistrationConfig(), environment);
+
+            doThrow(new RuntimeException("Oops")).when(badConsul).agentClient();
+
+            var now = Instant.now();
+            when(environment.currentInstant()).thenReturn(now);
+
+            var serviceInstance = ServiceInstance.fromServiceInfo(ServiceInfoHelper.buildTestServiceInfo())
+                    .withStatus(ServiceInstance.Status.UP);
+
+            assertThatThrownBy(() -> serviceToFail.register(serviceInstance))
+                    .isInstanceOf(RegistrationException.class)
+                    .hasMessageStartingWith("Unable to register service " + serviceInstance.getServiceName() + ", id ")
+                    .hasMessageEndingWith(" with Consul after 60 attempts");
+
+            assertThat(serviceToFail.registeredService.get()).isNull();
+        }
+
+        @Test
+        void shouldRegisterAndReturnRegisteredInstance_OverridingDomainIfConfiguredToDoSo() {
+            var now = Instant.now();
+            when(environment.currentInstant()).thenReturn(now);
+
+            var serviceInstance = ServiceInstance.fromServiceInfo(ServiceInfoHelper.buildTestServiceinfoWithHostName("example.com"))
+                    .withStatus(ServiceInstance.Status.UP);
+            config.setDomainOverride("test");
+
+            service.register(serviceInstance);
+
+            var services = consul.catalogClient().getService(serviceInstance.getServiceName()).getResponse();
+
+            var hostName = services.stream().map(CatalogService::getServiceAddress).findFirst();
+            assertThat(hostName).isPresent().get().isEqualTo("example.test");
+        }
+    }
+
+    @Nested
+    class UpdateStatus {
+
+        @Test
+        void shouldReturnNullIfNotRegistered() {
+            assertThat(service.updateStatus(ServiceInstance.Status.UP)).isNull();
+        }
+
+        @Test
+        void shouldReturnRegisteredInstanceUntouchedWhenRegistered() {
+            var serviceInstance = ServiceInstance.builder().build();
+            service.registeredService.set(serviceInstance);
+            assertThat(service.updateStatus(ServiceInstance.Status.UP)).isEqualToComparingFieldByField(serviceInstance);
```

Comment:
Is `ServiceInstance` a `@`Value` class? If so we can simply use `isEqualTo`

Reply:
No `@Getter` and `@Builder(toBuilder = true)`